### PR TITLE
For #12829:  Add create account option on the the sign in screen.

### PR DESCRIPTION
**Pull request by mcarare**
### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

File: app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -1138,6 +1138,10 @@
     <string name="sign_in_with_camera">Sign in with your camera</string>
     <!-- Text shown for settings option for sign with email -->
     <string name="sign_in_with_email">Use email instead</string>
+    <!-- Text shown for settings option for sign with email -->
```
Comment (ekager) [inline app/src/main/res/values/strings.xml:1141]: nit - let's add to the string description that Firefox is intentionally hardcoded?

Comment (ekager) [reply app/src/main/res/values/strings.xml:1141]: Also I don't think these strings descriptions are right? 🤔 Let's update those too!

File: app/src/main/java/org/mozilla/fenix/settings/account/TurnOnSyncFragment.kt [hunk 2]
```diff
@@ -89,6 +100,21 @@ class TurnOnSyncFragment : Fragment(), AccountObserver {
...
+        val spanEnd = spanStart + createAccountActionText.length
+
+        view.createAccount.apply {
+            text = fullText
+            addUnderline(
+                spanStart,
+                spanEnd,
+                Spanned.SPAN_EXCLUSIVE_EXCLUSIVE
+            )
+            setOnClickListener(createAccountClickListener)
```
Comment (ekager) [inline app/src/main/java/org/mozilla/fenix/settings/account/TurnOnSyncFragment.kt:116]: nit : could we just do this here?
```suggestion
            setOnClickListener { navigateToPairWithEmail() }
```

**Review by ekager — APPROVED** [review]
R+ with a few nits! Thanks!

File: app/src/main/res/values/strings.xml (hunk 1 above)
Comment (ekager) [reply app/src/main/res/values/strings.xml]: Oh and we probably also want to add what the substitution in the string is about generally

File: app/src/main/java/org/mozilla/fenix/settings/account/TurnOnSyncFragment.kt (hunk 2 above)
Comment (mcarare) [reply app/src/main/java/org/mozilla/fenix/settings/account/TurnOnSyncFragment.kt:141]: We could do that, but I intentionally made this like that, so any future change would be made in `createAccountClickListener` not inside the click listener setter. 
Also,  it is more easily readable that it is not a mistake, setting the same action on 2 different views.

**Review by ekager — APPROVED** [review]

**Pull request merged**